### ISSUE TITLE
A dataset that contains metadata sent by TRSA client  cannot be published (trsa-api branch)

### ISSUE THREAD

**Issue by akio-sone** [uncch-rdmc/dataverse#79]
![failed-to-publish-error-message_2021-05-03](https://user-images.githubusercontent.com/3188102/118518211-e59e2c00-b705-11eb-8b03-fd4d1fdafaee.png)
[dataverse-bug_cannot-publish.log](https://github.com/OdumInstitute/dataverse/files/6495050/dataverse-bug_cannot-publish.log)

Thank you for contributing an issue to the Dataverse Project! If this is a bug report, please let us know when the issue occurs, which page it occurs on, to whom it occurs, and which version of Dataverse you're using. If this is a feature request, please let us know what you'd like to see and give us some context - what kind of user is the feature intended for, the relevant use cases, and what inspired the request? No matter the issue, screenshots are always welcome.

**Comment by akio-sone** [comment]
Excerpt from server.log
```
[2021-05-12T05:52:08.781+0000] [Payara 5.2020.6] [INFO] [] [edu.harvard.iq.dataverse.engine.command.impl.FinalizeDatasetPublicationCommand] [tid: _ThreadID=208 _ThreadName=__ejb-thread-pool1] [timeMillis: 1620798728781] [levelValue: 800] [[
  Finalizing publication of the dataset doi:10.33563/FK2/1TZJUN]]

[2021-05-12T05:52:08.816+0000] [Payara 5.2020.6] [INFO] [] [edu.harvard.iq.dataverse.util.FileUtil] [tid: _ThreadID=208 _ThreadName=__ejb-thread-pool1] [timeMillis: 1620798728816] [levelValue: 800] [[
  successfully validated DataFile 21; checksum 0cdecbb13e886e16edba9786b5708478]]

[2021-05-12T05:52:08.819+0000] [Payara 5.2020.6] [INFO] [] [edu.harvard.iq.dataverse.util.FileUtil] [tid: _ThreadID=208 _ThreadName=__ejb-thread-pool1] [timeMillis: 1620798728819] [levelValue: 800] [[
  successfully validated DataFile 22; checksum 24b7fe60841efd1fdec247d3ce8cc31e]]

[2021-05-12T05:52:08.820+0000] [Payara 5.2020.6] [INFO] [] [edu.harvard.iq.dataverse.util.FileUtil] [tid: _ThreadID=208 _ThreadName=__ejb-thread-pool1] [timeMillis: 1620798728820] [levelValue: 800] [[
  Failed to open datafile id 23 for reading]]

[2021-05-12T05:52:08.832+0000] [Payara 5.2020.6] [WARNING] [] [edu.harvard.iq.dataverse.DatasetServiceBean] [tid: _ThreadID=208 _ThreadName=__ejb-thread-pool1] [timeMillis: 1620798728832] [levelValue: 900] [[
  CommandException caught when executing the asynchronous portion of the Dataset Publication Command.]]

[root@3418d80c141c logs]# 

```

**Comment by akio-sone** [comment]
This validation failure happens due to a new feature, "Datafiles Validation when Publishing Datasets", of version 5.0 (2020-08-18).  This feature can be turned off by an Native API call.  Since the granularity of this validation is not Datafile-wise but Dataset-wise, to programmatically avoid this issue is to add a new validation-exemption-condition for TRSA-uploaded cases by using the column of `trsa_id` of `dataset` table.  More specifically, `Dataset#isTrsaCoupled` method can be added to the following validation-exemption-conditions (line 76 of `FinalizeDatasetPublicationCommand` class) to avoid the validation step:
```
if (theDataset.getLatestVersion().getVersionState() != RELEASED
                && theDataset.getLatestVersion().getMinorVersionNumber() != null
                && theDataset.getLatestVersion().getMinorVersionNumber().equals((long) 0)
                && ctxt.systemConfig().isDatafileValidationOnPublishEnabled()) {
```

**Comment by akio-sone** [comment]
The above fix was tested with a local branch `trsa-api-v5.5-i79` and was committed to `trsa-api-v5.5` branch. This issue will be closed when cumulative changes to `trsa-api-v5.5` are merged into `trsa-api` branch.
